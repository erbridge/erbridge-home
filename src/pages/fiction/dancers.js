import PropTypes from 'prop-types';
import React, { Component } from 'react';

import './dancers.scss';

export class PoemWithSquare extends Component {
  static propTypes = {
    normalLines: PropTypes.arrayOf(PropTypes.string).isRequired,
    squareLines: PropTypes.arrayOf(PropTypes.string).isRequired,
    squareStartCharacter: PropTypes.number.isRequired,
    squareStartLine: PropTypes.number.isRequired,
  };

  render() {
    const {
      normalLines,
      squareLines,
      squareStartCharacter,
      squareStartLine,
    } = this.props;

    const topLines = normalLines.slice(0, squareStartLine);
    const middleLines = normalLines.slice(
      squareStartLine,
      squareStartLine + squareLines.length,
    );
    const bottomLines = normalLines.slice(squareStartLine + squareLines.length);

    const fontSize = 18;
    const lineHeight = fontSize * 1.5;

    return (
      <div className="PoemWithSquare">
        {topLines.map((line, i) => (
          <div key={i}>{line}</div>
        ))}
        <div className="PoemWithSquare__square-container">
          <div className="PoemWithSquare__before-square">
            {middleLines.map((line, i) => (
              <div key={i}>{line.substr(0, squareStartCharacter)}</div>
            ))}
          </div>
          <div
            className="PoemWithSquare__square"
            style={{ width: squareLines.length * lineHeight }}
          >
            {squareLines.map((line, i) => (
              <div key={i}>{line}</div>
            ))}
          </div>
          <div className="PoemWithSquare__after-square">
            {middleLines.map((line, i) => (
              <div key={i}>{line.substr(squareStartCharacter)}</div>
            ))}
          </div>
        </div>
        {bottomLines.map((line, i) => (
          <div key={i}>{line}</div>
        ))}
      </div>
    );
  }
}
