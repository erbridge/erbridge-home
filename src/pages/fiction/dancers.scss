@import '../../styles/_colours.scss';

.PoemWithSquare {
  margin: 1em 0;

  // FIXME: This is here to work around line wrapping on small screens.
  //        Probably better to wrap the contents of all the poetry in this
  //        sort of thing, but beware of making summaries long lines.
  overflow: auto;
  white-space: nowrap;
}

.PoemWithSquare__square-container {
  display: flex;
}

.PoemWithSquare__before-square {
  text-align: justify;
  text-align-last: justify;
  text-justify: distribute;
}

.PoemWithSquare__square {
  margin: 0 14px; // Measured in relation to font size and line spacing.
  text-align: justify;
  text-align-last: justify;
  text-justify: distribute;
  font-style: italic;
  color: $dark-blue;
}
