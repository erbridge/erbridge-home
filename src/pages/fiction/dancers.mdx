import MarkdownAudio from '../../components/MarkdownAudio';

import { PoemWithSquare } from './dancers';

export const meta = {
  title: 'Dancers',
  categories: ['poetry'],
  sortOrder: 11,
};

<MarkdownAudio src="/assets/fiction/dancers.mp3" />

<!-- prettier-ignore-start -->
Cold skin, warm heart.<br />
On a pedestal, exposed. Seen.<br />
Curves and motion examined.<br />
Eyes and minds, dancing the lines.<br />
Free flowing.<br />
Flowing free.
<!-- prettier-ignore-end -->

<PoemWithSquare
  normalLines={[
    'Merci. It starts with a kiss.',
    'Turns taken. Rules broken.',
    'A line from one hand to the next.',
    'Followed. Inspired. Ignored.',
    'Liberated.',
    'Five gestures.',
    'One movement.',
  ]}
  squareLines={['the square', 'divides us', 'reflects us', 'connects us']}
  squareStartCharacter={6}
  squareStartLine={1}
/>

<!-- prettier-ignore-start -->
Cold skin, warm heart.<br />
On a pedestal, stripped bare.<br />
Transformed.
<!-- prettier-ignore-end -->

> If read aloud, each line of the square (in teal) should be read by a second
> reader at the same time as the line it interrupts.
