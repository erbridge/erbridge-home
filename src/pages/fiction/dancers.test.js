import { shallow } from 'enzyme';
import React from 'react';

import { PoemWithSquare } from './dancers';

it('renders correctly', () => {
  expect(
    shallow(
      <PoemWithSquare
        normalLines={['this', 'is', 'a', 'test']}
        squareLines={['a', 'square']}
        squareStartCharacter={1}
        squareStartLine={1}
      />,
    ),
  ).toMatchSnapshot();
});
