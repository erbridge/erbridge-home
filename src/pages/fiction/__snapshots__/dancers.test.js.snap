// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="PoemWithSquare"
>
  <div
    key="0"
  >
    this
  </div>
  <div
    className="PoemWithSquare__square-container"
  >
    <div
      className="PoemWithSquare__before-square"
    >
      <div
        key="0"
      >
        i
      </div>
      <div
        key="1"
      >
        a
      </div>
    </div>
    <div
      className="PoemWithSquare__square"
      style={
        Object {
          "width": 54,
        }
      }
    >
      <div
        key="0"
      >
        a
      </div>
      <div
        key="1"
      >
        square
      </div>
    </div>
    <div
      className="PoemWithSquare__after-square"
    >
      <div
        key="0"
      >
        s
      </div>
      <div
        key="1"
      />
    </div>
  </div>
  <div
    key="0"
  >
    test
  </div>
</div>
`;
